app/assets/javascripts/components/story/CollapsedStory/index.js

```diff
@@ -23,39 +27,177 @@ const storyClassName = (story, additionalClassname = '') => {
       'Story--estimated': !isStoryNotEstimated
     },
     additionalClassname
-  );
-};
-
-export const CollapsedStory = ({ onToggle, story, updateCollapsedStory, project, className, title }) =>
-  <div
-    className={storyClassName(story, className)}
-    onClick={onToggle}
-    title={title}
-  >
-    <StoryPopover story={story}>
-      <div className='Story__icons-block'>
-        <StoryIcon storyType={story.storyType} />
-        <CollapsedStoryEstimate estimate={story.estimate} />
-        <StoryDescriptionIcon description={story.description} />
-      </div>
-    </StoryPopover>
-
-    <CollapsedStoryInfo story={story} />
-
-    <CollapsedStoryStateActions story={story}
-      onUpdate={(newAttributes) =>
-        updateCollapsedStory(story.id, project.id, newAttributes)}
-    />
-  </div>
+  )
+}
+
+const type = {
+  story: 'STORY'
+}
+
+const calculatePosition = (aboveStory, bellowStory, storiesArray, index) => {
+  if (bellowStory === undefined) {
+    return (Number(storiesArray[index].position) + 1)
+  }else if (aboveStory !== undefined && bellowStory !== undefined) {
+    return (Number(bellowStory.position) + Number(aboveStory.position)) / 2;
+  }else if (aboveStory === undefined) {
+    return (Number(storiesArray[index].position) - 1);
+  }
+}
+
+const getNewPosition = (item, story, storiesArray, index) => {
+  let aboveStory;
+  let bellowStory;
+
+  if(item.index > index) {
+    aboveStory = storiesArray[index - 1]
+    bellowStory = story
+  } else {
+    bellowStory = storiesArray[index + 1]
+    aboveStory = story
+  }
+
+  return calculatePosition(aboveStory, bellowStory, storiesArray, index)
+}
```

Comment:
You don't need these two variables (`aboveStory` and `bellowStory `), you can do:
```js
const getNewPosition = (item, story, storiesArray, index) => {
  if(item.index > index) {
    return calculatePosition(storiesArray[index - 1], story, storiesArray, index);
  }
    
  return calculatePosition(story, storiesArray[index + 1], storiesArray, index);
}
```

Reply:
Looks nice, thanks